src/NamespaceLabeler.ts

```diff
@@ -0,0 +1,68 @@
+﻿import { ConfigReader, AddonConfig } from "./ConfigReader"
+import { setTimeout } from "timers";
+import k8s = require("@kubernetes/client-node");
+import { logger } from "./LoggerWrapper";
+import { V1Namespace } from "@kubernetes/client-node";
+
+export class NamespaceLabeler {
+    private delay: number = 15 * 60 * 1000
+
+    private loop(delay?: number) {
+        if (delay != null || delay == 0) {
+            this.delay = delay*60*1000;
+        }
+
+        setTimeout(async () => {
+            logger.info("relooping")
+            await this.labelNamespace();
+        }, this.delay)
+    }
+
+    private async labelNamespace():Promise<any> {
+        return ConfigReader.ReadConfig()
+            .then((config:AddonConfig) => {
+                logger.info(`got config ${JSON.stringify(config)}`);
+                const kc = new k8s.KubeConfig();
+                kc.loadFromDefault();
+                const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+                k8s.V1Namespace
+                return k8sApi.listNamespace().then((result) => {
+                    logger.info(`got namespace list ${JSON.stringify(result)}`);
+                    const namespaceList = result.body.items;
+                    namespaceList.forEach((item: V1Namespace) => {
+                        const patchPayload = item;
+                        if (patchPayload.metadata.labels == null) {
+                            patchPayload.metadata.labels = {};
+                        }
+                        patchPayload.metadata.labels["app-monitoring"] =
+                            config.excludedNamespaces.indexOf(item.metadata.name) < 0 ?
+                            "enable" : "disabled";
```

Comment:
Does this mean we're adding labels app-monitoring=false on namespaces that we don't care about? Is that the intention? Can we just not add anything for those namespaces or remove the label for them?

Reply:
yeah we can , i was more looking to see the result when executing it, i'll fix it